module/VuFind/src/VuFind/Form/Handler/Email.php

```diff
@@ -0,0 +1,209 @@
+     *
+     * @var RendererInterface
+     */
+    protected $viewRenderer;
+
+    /**
+     * Main config
+     *
+     * @var Config
+     */
+    protected $mainConfig;
+
+    /**
+     * Mailer
+     *
+     * @var Mailer
+     */
+    protected $mailer;
+
+    /**
+     * Contructor
+     *
+     * @param RendererInterface $viewRenderer View renderer
+     * @param Config            $config       Main config
+     * @param Mailer            $mailer       Mailer
+     */
+    public function __construct(
+        RendererInterface $viewRenderer,
+        Config $config,
+        Mailer $mailer
+    ) {
+        $this->viewRenderer = $viewRenderer;
+        $this->mainConfig = $config;
+        $this->mailer = $mailer;
+    }
+
+    /**
+     * Gets data from submitted form and process them.
+     * Returns array with keys: (bool) success - mandatory, (string) errorMessages,
+     * (string) successMessage, (string) errorMessagesDetailed - used to log an
+     * error
+     *
+     * @param \VuFind\Form\Form                     $form   Submitted form
+     * @param \Laminas\Mvc\Controller\Plugin\Params $params Request params
+     * @param ?\VuFind\Db\Row\User                  $user   Authenticated user
+     *
+     * @return array
+     */
+    public function handle(
+        \VuFind\Form\Form $form,
+        \Laminas\Mvc\Controller\Plugin\Params $params,
+        ?\VuFind\Db\Row\User $user = null
+    ): array {
+        $fields = $form->mapRequestParamsToFieldValues($params->fromPost());
+        $emailMessage = $this->viewRenderer->partial(
+            'Email/form.phtml',
+            compact('fields')
+        );
+
+        [$senderName, $senderEmail] = $this->getSender($form);
+
+        $replyToName = $params->fromPost(
+            'name',
+            $user ? trim($user->firstname . ' ' . $user->lastname) : null
+        );
+        $replyToEmail = $params->fromPost(
+            'email',
+            $user ? $user->email : null
+        );
+        $recipients = $form->getRecipient($params->fromPost());
+        $emailSubject = $form->getEmailSubject($params->fromPost());
+
+        $sendSuccess = true;
+        $errorsDetailed = [];
+        $success = false;
+        foreach ($recipients as $recipient) {
+            [$success, $errorMsg] = $this->sendEmail(
+                $recipient['name'],
+                $recipient['email'],
+                $senderName,
+                $senderEmail,
+                $replyToName,
+                $replyToEmail,
+                $emailSubject,
+                $emailMessage
+            );
+
+            $sendSuccess = $sendSuccess && $success;
+            if (!$success) {
+                $errorsDetailed[]  = $errorMsg;
+            }
+        }
+        return [
+            'success' => $success,
+            'errorMessagesDetailed' => $errorsDetailed,
+            'errorMessages'
+                => $success ? [] : ['Could not send e-mail with your feedback.'],
```

Comment:
I don't think the user should know what failed and how. It's better to log the details and give the user just a generic "Could not process your feedback. Please try again later." or such.